src/core/ComputedAttribute.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react'
+import { BufferAttribute, BufferGeometry } from 'three'
+
+type Props = {
+  compute: (geometry: BufferGeometry) => BufferAttribute
+  name: string
+}
+
+/**
+ * Used exclusively as a child of a BufferGeometry.
+ * Computes the BufferAttribute by calling the `compute` function
+ * and attaches the attribute to the geometry.
+ */
+export const ComputedAttribute: React.FC<Props> = ({ compute, name, ...props }) => {
+  const [bufferAttribute] = React.useState(() => new BufferAttribute(new Float32Array(0), 1))
+  const primitive = React.useRef<BufferAttribute>(null)
+
+  React.useLayoutEffect(() => {
+    if (primitive.current) {
+      // @ts-expect-error parent doesnt'e exist on BufferAttribute
+      const attr = compute(primitive.current.parent as BufferGeometry)
```

Comment:
Did you mean `primitive.current.__r3f.parent`?

Reply:
I found that doesn't consistently work, for some reason @drcmda

Reply:
Ah well i though this might be what's breaking the CSB

Reply:
@gsimone @FarazzShaikh it should be __r3f.parent but it can well be that this is only fully consistent in v8. in v8 this will break, it won't write parent onto non scene objects, only __r3f.parent. but it's ok to merge if it works now.

Reply:
It doesn't work consinstently with any of the two, but I think I can be cheeky and have both methods, just taking the one that's defined 😅

Reply:
Until we move to v8, that is